docs/beaver-builder/troubleshooting/common-issues/error-settings-not-saved.md

```diff
@@ -1,22 +1,49 @@
+
+In WordPress, user roles, (such as Administrator or Editor, contain a set of [capabilities](https://wordpress.org/support/article/roles-and-capabilities/), which define the particular tasks that users can perform. Custom roles, such as the Shop Manager in WooCommerce or custom roles that you create, have a custom set of capabilities. 
+
+One capability is called [`unfiltered_html`](https://wordpress.org/support/article/roles-and-capabilities/#unfiltered_html). User roles with this capability are allowed to enter sensitive code with tags such as `<script>` or `<iframe>` into post titles and content. By default, only the WordPress Administrator and Editor roles have this capability, while only the Super Admin role has it on multisite installations. Here's [more information about this capability](https://publishpress.com/blog/unfiltered-html-permission-wordpress/). The WooCommerce [Shop Manager role](https://docs.woocommerce.com/document/roles-capabilities/#section-2) does not have the `unfiltered_html` capability, and other custom roles may not have it, even though users with this role can perform many other types of administrative tasks.
 
 If you see this error, you'll not be able to save without removing the sensitive code or employing one of the following workarounds:
 
 * Log in with a user role that includes the `unfiltered_html` capability, such as WordPress Admin or Editor.
-* Use a WordPress plugin that gives you the ability to assign different capabilities to roles, such as the [User Role Editor](https://wordpress.org/plugins/user-role-editor/) plugin.
+* Use a WordPress plugin that gives you the ability to assign different capabilities to roles, such as the [User Role Editor](https://wordpress.org/plugins/user-role-editor/) plugin, and assign the `unfiltered_html` capability to the user role you want to have it.
+
+See [our blog post](https://www.wpbeaverbuilder.com/wordpress-user-roles/) for basics about WordPress user roles.
+
+## The DISALLOW_UNFILTERED_HTML setting is in use
+
+In this case, you'll see the following version of the error.
+
+> Settings could not be saved.  
+> Sensitive code (with tags such as `<iframe>` or `<script>`) is not allowed because the site's wp-config.php file contains the DISALLOW_UNFILTERED_HTML setting.
+
+This happens when the WordPress *wp-config.php* file has the following setting:
+
+```
+define( 'DISALLOW_UNFILTERED_HTML', true );
+```
+
+This setting prevents all users, regardless of their user role capabilities, from being able to insert scripts in the title and body of posts and pages. 
+
+In this case, the alternatives are:
 
-See [our blog post](https://www.wpbeaverbuilder.com/wordpress-user-roles/) for basics about WordPress user roles.
\ No newline at end of file
+* Remove the `DISALLOW_UNFILTERED_HTML` setting from the *wp-config.php* file for your WordPress installation.
+* Remove the sensitive code from the title or content of your Beaver Builder layout and save your layout.
```

Comment:
@nancyhild 

Users can add the following filter to their child theme's functions.php file and this will give all user roles `unfiltered_html`.

```php
add_filter( 'fl_builder_ui_js_config', function( $config ) {
    $config['userCaps']['unfiltered_html'] = true;
    return $config;
} ,10);
```

The snippet below will give a specific user role (in this case the WooCommerce **Shop Manager** role `unfiltered_html`.

```php
add_filter( 'fl_builder_ui_js_config', function( $config ) {
 	$user = wp_get_current_user();
	$role = $user->roles[0];
 	if ($role == "shop_manager") {
    	$config['userCaps']['unfiltered_html'] = true;
 	}
    return $config;
} ,10);
```

or finally, both the **Shop Manager** and **Author** roles `unfiltered_html`.

```php
add_filter( 'fl_builder_ui_js_config', function( $config ) {
 	$user = wp_get_current_user();
	$role = $user->roles[0];
 	if ($role == "shop_manager" || $role == "author") {
    	$config['userCaps']['unfiltered_html'] = true;
 	}
    return $config;
} ,10);
```

Note: This will only give the user roles above `unfiltered_html` in the BB UI, not in WordPress.